phpunit/class-rest-styles-controller-test.php

```diff
@@ -0,0 +1,287 @@
+		$this->assertCount( 1, $data[1]['deps'] );
+		$this->assertEquals( array( 'dependency3' ), $data[1]['deps'] );
+		$this->assertCount( 1, $data[2]['deps'] );
+		$this->assertEquals( array( 'dependency4' ), $data[2]['deps'] );
+	}
+
+	/**
+	 * Test single style.
+	 */
+	public function test_get_item() {
+		wp_set_current_user( self::$admin_id );
+		$request  = new WP_REST_Request( 'GET', '/__experimental/styles/' . self::$style_handle );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertEquals( self::$style_handle, $data['handle'] );
+		$this->assertEquals( home_url( '/test.css' ), $data['src'] );
+		$this->assertEquals( home_url( '/test.css' ), $data['url'] );
+	}
+
+	/**
+	 * Create item test.
+	 */
+	public function test_create_item() {
+		// Not testable.
+	}
+
+	/**
+	 * Update item test.
+	 */
+	public function test_update_item() {
+		// Not testable.
+	}
+
+	/**
+	 * Delete item test.
+	 */
+	public function test_delete_item() {
+		// Not testable.
+	}
+
+	/**
+	 * Test prepare item.
+	 */
+	public function test_prepare_item() {
+		wp_set_current_user( self::$admin_id );
+		$request  = new WP_REST_Request( 'GET', '/__experimental/styles/' . self::$style_handle );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertArrayHasKey( 'handle', $data );
+		$this->assertArrayHasKey( 'src', $data );
+		$this->assertArrayHasKey( 'url', $data );
+		$this->assertArrayHasKey( 'args', $data );
+		$this->assertArrayHasKey( 'ver', $data );
+		$this->assertArrayHasKey( 'extra', $data );
+		$this->assertArrayHasKey( 'textdomain', $data );
+		$this->assertArrayHasKey( 'translations_path', $data );
+		$this->assertArrayHasKey( 'deps', $data );
+		$links = $response->get_links();
+		$this->assertArrayHasKey( 'self', $links );
+		$this->assertArrayHasKey( 'collection', $links );
+		$this->assertArrayHasKey( 'deps', $links );
+	}
+
+	/**
+	 * Test schema.
+	 */
+	public function test_get_item_schema() {
+		$request  = new WP_REST_Request( 'OPTIONS', '/__experimental/styles' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertArrayHasKey( 'schema', $data );
+		$this->assertArrayHasKey( 'properties', $data['schema'] );
+		$properties = $data['schema']['properties'];
+		$this->assertEquals( 9, count( $properties ) );
+		$this->assertArrayHasKey( 'handle', $properties );
+		$this->assertArrayHasKey( 'src', $properties );
+		$this->assertArrayHasKey( 'url', $properties );
+		$this->assertArrayHasKey( 'args', $properties );
+		$this->assertArrayHasKey( 'ver', $properties );
+		$this->assertArrayHasKey( 'extra', $properties );
+		$this->assertArrayHasKey( 'textdomain', $properties );
+		$this->assertArrayHasKey( 'translations_path', $properties );
+		$this->assertArrayHasKey( 'deps', $properties );
+	}
+
+	/**
+	 * Tests set up.
+	 */
+	public function setUp() {
```

Comment:
Nit pic, setup normally appears at the top of the class.

Reply:
👍